### PR TITLE
Create GraphQL request flow diagram

### PR DESCRIPTION
**Pull request by tomekpapiernik**
**Description**

Changes proposed in this pull request:

- Added GraphQL request flow diagram and description. 

**Related issue(s)**
#2453

### CODE REVIEW

**Comment by piotrmsc** [comment]
It's token validation not authorization, and diagram should mention "return user permissions to requested resource"

**Comment by tomekpapiernik** [comment]
> It's token validation not authorization, and diagram should mention "return user permissions to requested resource"

Fix'd

File: docs/security/docs/03-03-graphql-request-flow.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+This diagram illustrates the request flow for the Console Backend Service, which uses a custom [GraphQL](https://graphql.org/) implementation:
+
+![GraphQL request flow](./assets/002-graphql-request-flow.svg)
+
+1. The user sends a request with ID token to the GraphQL application.
+2. The GraphQL application validates the user token and extracts user data required to perform [Subject Access Review](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#checking-api-access) (SAR).
+3. The [Kubernetes API Server](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/) performs SAR.
+4. Based on the results of SAR, the Kubernetes API Server informs the GraphQL application whether the user can perform the requested [GraphQL action](#details-graphql-available-graphql-actions).
+5. Based on the information provided by the Kubernetes API Server, the GraphQL application return an appropriate response to the user.
```
Comment (bszwarc) [inline docs/security/docs/03-03-graphql-request-flow.md]:
```suggestion
5. Based on the information provided by the Kubernetes API Server, the GraphQL application returns an appropriate response to the user.
```

File: docs/security/docs/03-03-graphql-request-flow.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+title: GraphQL request flow
+type: Details
+---
+
+This diagram illustrates the request flow for the Console Backend Service, which uses a custom [GraphQL](https://graphql.org/) implementation:
+
+![GraphQL request flow](./assets/002-graphql-request-flow.svg)
+
+1. The user sends a request with ID token to the GraphQL application.
```
Comment (bszwarc) [inline docs/security/docs/03-03-graphql-request-flow.md]:
```suggestion
1. The user sends a request with an ID token to the GraphQL application.
```

File: docs/security/docs/03-03-graphql-request-flow.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+---
+title: GraphQL request flow
+type: Details
+---
+
+This diagram illustrates the request flow for the Console Backend Service, which uses a custom [GraphQL](https://graphql.org/) implementation:
```
Comment (bszwarc) [inline docs/security/docs/03-03-graphql-request-flow.md]:
```suggestion
This diagram illustrates the request flow for the Console Backend Service which uses a custom [GraphQL](https://graphql.org/) implementation:
```

**Pull request merged**